jujugui/static/gui/src/app/app.js

```diff
@@ -1240,13 +1240,14 @@ YUI.add('juju-gui', function(Y) {
     */
     _generateSocketUrl: function(callback) {
       this.jem = null;
-      if (this.get('jemUrl')) {
+      if (this.get('jemURL')) {
         var bakery = new Y.juju.environments.web.Bakery({
           webhandler: new Y.juju.environments.web.WebHandler(),
           interactive: this.get('interactiveLogin'),
           serviceName: 'jem'
         });
-        this.jem = new window.jujulib.environment(this.get('jemUrl'), bakery);
+        this.jem = new window.jujulib.environment(
+            this.get('jemURL'), this.get('jemAPIPath'), bakery);
```

Comment:
I imagine that once we move away from YUI we'll always be fetching from the juju_config anyways :) Using the ATTRS just provides us with a way to add getters/setters/docs etc. So I'm not too picky about switching them INTO YUI, just that if we are using it we should define it.